src/cljfx/context.clj

```diff
@@ -90,17 +92,27 @@ Possible reasons:
      ::direct-deps @*direct-deps
      ::key-deps @*key-deps}))
 
-(declare sub)
+(declare sub register-cache-entry)
+
+(def ^:dynamic *processing-dirty* #{})
 
 (defn- sub-from-dirty [context *cache cache sub-id]
   (let [dirty-sub (lookup cache [::dirty sub-id])
-        deps (::direct-deps dirty-sub)
-        unbound-context (unbind context)]
-    (if (every? #(= (get deps %) (apply sub unbound-context %)) (keys deps))
+        unbound-context (unbind context)
+        key-deps (reduce
+                   (fn [key-deps [dep-id dep-v]]
+                     (let [v (-> (register-cache-entry unbound-context *cache @*cache dep-id)
+                                 (lookup dep-id))]
+                       (if (not= dep-v (::value v))
+                         (reduced nil)
+                         (into key-deps (::key-deps v)))))
+                   (::key-deps dirty-sub)
```

Comment:
Are you sure we need to keep key deps of dirty sub? it may introduce unnecessary dependency in a scenario where we had one key dep, and then changed it for another. I tried replacing this line with `#{}` and it seems to work...